content/docs/reference/design-docs/editoast-errors/_index.en.md

```diff
@@ -0,0 +1,431 @@
+- Consequently, we hardly can declare in the OpenApi file what errors an endpoint precisely returns, degrading the editoast API quality
+- The frontend still requires editoast to declare all its errors though, to ensure they are translated properly. To achieve that we dynamically collect each `EditoastError` using the crate `inventory`. All the error descriptions collected are then transformed into OpenAPI schemas procedurally. On top of being a Rust antipattern (collecting state in proc-macros), this is complex to maintain on both editoast and frontend sides.
+  - Not having each endpoint linked to the list of errors it can raise, also prevents the frontend easily handling errors properly.
+- It's still unclear how we should expose errors from Core.
+
+# Goals
+
+- Have a clear separation between logically distinct errors.
+- Dispose of a way to actually match on errors when they occur deeper in the stack
+- Tie the errors to the endpoint they originate from in the OpenAPI.
+- Separate the error definition and their serialization.
+- Establish how we want to forward Core's errors.
+- End the `InternalError` hegemony ⚔️
+
+# Constraints
+
+- Keep the same error format (for backward compatibility reasons to avoid involving the frontend too much).
+- The error must live until it is handled, conversion to our standard error format only happens in the response serialization.
+- Errors must implement `std::error::Error`.
+- Errors must be composable. This will typically be handled by `thiserror`'s `#[from]` attribute.
+- Error variants must be shareable to ensure the deduplication of error kinds. For example, let's say we have two functions `get_infra(id: usize)` and `rename_infra(id: usize, name: String)`. Both functions error types have to include a variant describing the error case of an infrastructure not being found by its ID. However, we can't duplicate something like `InfraNotFound { id: usize }` in both error types as this leads to two different error paths describing the same error case. This is especially problematic for error translation keys. We need to be able to define an error `InfraNotFound` and include it in both error types.
+- Each endpoint must provide all its error cases in the OpenAPI. (How the frontend will consume them is another problem that we'll have to deal with.)
+- As for OSRD errors, the `context` field of the error is populated in the views.
+- Errors can be handled in a generic manner (for situations where it makes some sense to do so).
+  - I.e.: some form of downcasting is available.
+
+# New system
+
+- Rely on `thiserror` everywhere.
+- Keep the `trait EditoastError` but only implement it for errors defined in `views`.
+  - Since it is only used in `views` now, let's rename it to `trait ViewError`.
+  - Let's rename the macro to `derive(ViewError)` as well.
+- Create a macro `derive(ViewError)` which interfaces with `derive(thiserror::Error)`. We can start from the former `derive(EditoastError)`.
```

Comment:
Maybe a little inconsistency here since L44 tells about renaming the macro and L45 about creating a new one (based on the old one).

Reply:
Fixed in [`ab758a5` (#237)](https://github.com/OpenRailAssociation/osrd-website/pull/237/commits/ab758a5569319c2e4320c9cb272f301ae02de51a)